src/com/gfredericks/test/chuck/clojure_test.cljc

```diff
@@ -88,14 +88,13 @@
 (defmacro qc-and-report-exception
   [final-reports num-tests-or-options bindings & body]
   `(report-exception-or-shrunk
-     (let [num-tests-or-options# ~num-tests-or-options]
-       (apply tc/quick-check
-         (times num-tests-or-options#)
-         (prop/for-all ~bindings
-           (let [reports# (capture-reports ~@body)]
-             (swap! ~final-reports save-to-final-reports reports#)
-             (pass? reports#)))
-         (apply concat (options num-tests-or-options#))))))
+     (apply tc/quick-check
+            (times ~num-tests-or-options)
+            (prop/for-all ~bindings
+                          (let [reports# (capture-reports ~@body)]
+                            (swap! ~final-reports save-to-final-reports reports#)
+                            (pass? reports#)))
+            (apply concat (options ~num-tests-or-options)))))
```

Comment:
The reason there was a `let` clause at the top in the old code is that it lets you use `num-tests-or-options` in two places without inserting the code twice. This is general good-practice when writing a macro because otherwise if users have a nontrivial expression, it will be evaluated twice.

E.g., compare the expansions of `(doubler (inc 20))` with these two definitions:

```
(defmacro doubler
  [x]
  `(+ ~x ~x))

(defmacro doubler
  [x]
  `(let [x# ~x] (+ x# x#)))
```

Reply:
Now that I look closer, have you actually changed anything in this code, other than inlining `num-tests-or-options`? If not, then you should be able to just revert to the original code.

Reply:
Never thought of the double evaluation. Thanks for the tip.

Reply:
Code reverted by the way